src-vue/src/plugins/store.ts

```diff
@@ -0,0 +1,193 @@
+                        console.error(error);
+                        alert(error);
+                    });
+
+                _get_northstar_version_number(state);
+                return;
+            }
+
+            // Update northstar if it is outdated.
+            if (state.northstar_state === NorthstarState.MUST_UPDATE) {
+                // Updating is the same as installing, simply overwrites the existing files
+                let install_northstar_result = invoke("install_northstar_caller", { gamePath: state.game_path, northstarPackageName: ReleaseCanal.RELEASE });
+                state.northstar_state = NorthstarState.UPDATING;
+
+                await install_northstar_result.then((message) => {
+                    console.log(message);
+                })
+                    .catch((error) => {
+                        console.error(error);
+                        alert(error);
+                    });
+
+                _get_northstar_version_number(state);
+                return;
+            }
+
+            // Game is ready to play
+            if (state.northstar_state === NorthstarState.READY_TO_PLAY) {
+                // Show an error message if Origin is not running.
+                if (!state.origin_is_running) {
+                    ElNotification({
+                        title: 'Origin is not running',
+                        message: "Northstar cannot launch while you're not authenticated with Origin.",
+                        type: 'warning',
+                        position: 'bottom-right'
+                    });
+
+                    // If Origin isn't running, end here
+                    return;
+                }
+
+                let game_install = {
+                    game_path: state.game_path,
+                    install_type: state.install_type
+                } as GameInstall;
+                await invoke("launch_northstar_caller", { gameInstall: game_install })
+                    .then((message) => {
+                        console.log(message);
+                        // NorthstarState.RUNNING
+                    })
+                    .catch((error) => {
+                        console.error(error);
+                        alert(error);
+                    });
+                return;
+            }
+        }
+    }
+});
+
+/**
+ * This is called when application root component has been mounted.
+ * It invokes all Rust methods that are needed to initialize UI.
+ */
+async function _initializeApp(state: any) {
+    // @ts-ignore
+    const result: GameInstall = await invoke("find_game_install_location_caller")
+        .catch((err) => {
+            // Gamepath not found or other error
+            console.error(err);
+            alert(err);
+        });
+    state.game_path = result.game_path;
+    state.install_type = result.install_type as InstallType;
+
+    // Check installed Northstar version if found
+    await _get_northstar_version_number(state);
+}
+
+// TODO
```

Comment:
Still `TODO`? ^^

Reply:
Still TODO, yeah, I didn't import this feature from previous code

Reply:
Wait what exactly is missing here? ^^

Reply:
Copypasting your ancient code and adapting it to the new store.

Reply:
Done with 4fc36ee367cb7abb18277ee59a7acdcead66ff82
Does the method actually update both FlightCore AND Northstar? You might want to rename it if yes.

Reply:
Done with 4fc36ee367cb7abb18277ee59a7acdcead66ff82
Does the method actually update both FlightCore AND Northstar? You might want to rename it if yes.

Reply:
Mm, I might have imported unused code as well...

Reply:
Yeah, I think you copied a few things too many here ^^

`_checkForFlightCoreUpdates()` should really just check whether FlightCore is outdated IMO.

Reply:
Also I just now realised but why is `store.ts` in folder called `plugins`? ^^

That might confuse things with `Store`, imported from `tauri-plugin-store-api`

https://github.com/GeckoEidechse/FlightCore/blob/a25a4ad4fda8c84ed3fd355f9494b28025dd0bbc/src-ui/src/main.ts#L5

Reply:
Because stores aren't included in Vue projects by default, Vuex is a plugin to Vue :)

Reply:
Aight, but our `store` is really just an ephemeral store of current state, not to be confused with a persistent store to disk, right? ^^

Reply:
This is still a frontend store, which is meant to be temporary!

Reply:
(plus, I think the tauri Store app import is useless and will be removed)

Reply:
> This is still a frontend store, which is meant to be temporary!

I mean yeah. It's just the double use of the term is kinda confusing for someone going through the code I guess so I might update the nomenclature a bit post merge :P

> (plus, I think the tauri Store app import is useless and will be removed)

In current UI rewrite yeah, it does nothing. In the future it would persist the release channel selection (among potentially other things) to disk ^^

Reply:
Yeah, you'll just have to create `uiStore` and `persistentStore` variables if you want it to be more clear.

Reply:
Self-update checking implemented with dddbc93c3bab88a9d4ad3752cdfe35e15692e03f